### PR TITLE
Jb/events sorting

### PR DESCRIPTION
**Pull request by jbouzi12**
This PR updates all our events lists to ensure:
- they are sorted
- they only display upcoming events (I defined this as today or later)

### CODE REVIEW

File: src/hooks/use-event-data.js [hunk 1]
```diff
@@ -49,6 +49,21 @@ export const sampleEvents = [
     },
 ];
 
+export const removePastEvents = events => {
```
Comment (jestapinski) [inline src/hooks/use-event-data.js]: Super nit, but I noticed the `React` import on line 1 is no longer used. Should we remove it?

Comment (jbouzi12) [reply src/hooks/use-event-data.js]: Good catch 👍

**Comment by jbouzi12** [comment]
> LGTM, I took a look in Firefox and Edge to make sure everything was filtered appropriately and looked fine.
> 
> Not sure if there is any concern with UTC comparison like in the author off by one date issue. It looks like the dates in the sample are timezone aware and the compared date (today) is UTC, so maybe there could be an off by one issue? That might be worth a look, but not really sure.
> 
> Looks great!

One thing is that this sorting wont work on IE because it doesn't support `Array.sort()`. But I assumed that we won't be supporting IE because our target audience is much less likely to be on it.

The date disparity is a good point though. I'm going to update the comparison to ensure all dates used are in `UTC`.

File: src/hooks/use-event-data.js [hunk 2]
```diff
@@ -49,6 +49,21 @@ export const sampleEvents = [
...
+    return events.filter(
+        e => new Date(e.node_type_attributes.event_end) >= today
+    );
+};
+export const sortEvents = events =>
+    // create new list first so original does not get mutated
+    // this improves testability and reduces side-effects
+    [...events].sort((a, b) => {
+        const date1 = new Date(a.node_type_attributes.event_start).getTime();
+        const date2 = new Date(b.node_type_attributes.event_start).getTime();
```
Comment (greggb) [inline src/hooks/use-event-data.js]: Is the `getTime` necessary?

File: src/hooks/use-event-data.js [hunk 3]
```diff
@@ -49,6 +50,23 @@ export const sampleEvents = [
...
+    return events.filter(
+        e =>
+            createDateObject(new Date(e.node_type_attributes.event_end)) >=
+            today
+    );
+};
+export const sortEvents = events =>
+    // create new list first so original does not get mutated
+    // this improves testability and reduces side-effects
+    [...events].sort((a, b) => {
```
Comment (greggb) [inline src/hooks/use-event-data.js]: I knew we had spoken briefly about this, but seems like we could get sorted events with ` https://www.mongodb.com/api/event/all/1?sort=-node_type_attributes.event_start`

File: src/hooks/use-event-data.js [hunk 4]
```diff
@@ -49,6 +50,23 @@ export const sampleEvents = [
     },
 ];
 
+export const removePastEvents = events => {
+    const today = createDateObject(new Date());
+    return events.filter(
+        e =>
+            createDateObject(new Date(e.node_type_attributes.event_end)) >=
```
Comment (greggb) [inline src/hooks/use-event-data.js]: We should also filter on `status === 'published'`

File: src/components/dev-hub/event-list.js [hunk 5]
```diff
@@ -49,7 +50,9 @@ export const EventsListPreview = () => {
                 <Event key={event.url} event={event} />
             ))}
         </EventsPreview>
-    ) : null;
+    ) : (
+        <P>Check back later for upcoming events</P>
```
Comment (greggb) [inline src/components/dev-hub/event-list.js]: Do we need this to match the error handling on the events page?

**Pull request merged**